src/games/stendhal/client/GameScreen.java

```diff
@@ -591,6 +592,10 @@ public void paintComponent(final Graphics g) {
 		drawEmojis(g2d);
 
 		paintOffLineIfNeeded(g2d);
+
+		// Ask window manager to not skip frame drawing
+		Toolkit.getDefaultToolkit().sync();
```

Comment:
On my system (Fedora linux) rendering is skipping frames, especially when mouse cursor was not moving / not present over window. This improved it a bit, although frames intervals were still inconsistent, until I switched rendering to OpenGL.
Cannot say for sure if this could lead to lags on other systems / JDK versions, only that various sources on web recommended this for frame-based rendering of games in Java. On Windows 11 it didn't seem to have any impact as well as in Windows 10 - although I have Windows 10 only in virtual machine and performance was a bit poor, but seemed same whether sync was present or not.

Command documentation:
https://docs.oracle.com/javase/8/docs/api/java/awt/Toolkit.html#sync--